tokio/src/util/linked_list.rs

```diff
@@ -243,6 +229,53 @@ impl<L: Link> fmt::Debug for LinkedList<L, L::Target> {
     }
 }
 
+// ===== impl CountedLinkedList ====
+
+// Delegates operations to the base LinkedList implementation, and adds a counter to the elements
+// in the list.
+pub(crate) struct CountedLinkedList<L: Link, T> {
+    _list: LinkedList<L, T>,
+    _count: AtomicUsize,
```

Comment:
You have a mutable reference any time you modify it, so you don't need an atomic.
```suggestion
    _list: LinkedList<L, T>,
    _count: usize,
```
If you want an atomic, then you should place it outside of the mutex. In that case, the logic would need to go in `OwnedTasks` rather than in the linked list.

Reply:
You seem to have marked this as resolved, but it does not appear to have been addressed.

The point of using an atomic is to avoid locking the mutex when accessing the counter. However, because the atomic is inside the mutex, you are still locking the mutex when accessing the counter.

Reply:
My apologies marking this resolved without actually implementing the changes you had requested. It was not intentional, and I regret any inconvenience this may have caused. I will make sure to be more careful in the future to avoid similar mistakes. Thank you for your time and understanding.

Reply:
Thanks, it looks better now.

There's no need to apologize for something like this. I was not offended or unhappy about it — I just wanted to clarify what I meant in case my review comment was unclear.

Reply:
Makes sense, all good! 😄